src/components/ErrorPages/BrowserWarning.tsx

```diff
@@ -15,21 +27,49 @@ const BrowserWarning = () => {
         version: browser.version || "Unknown",
       };
     }
+    return null;
   }, []);
 
+  const [showUnsupportedBrowserDialog, setShowUnsupportedBrowserDialog] =
+    useState(true);
+
   if (!notSupported) {
     return null;
   }
 
   return (
-    <div className="fixed left-0 top-0 z-50 flex h-20 w-full items-center justify-center bg-gray-800 bg-opacity-60 text-center text-gray-300">
-      <div>
-        <h2 className="text-lg font-medium">{t("unsupported_browser")}</h2>
-        <p className="text-sm">
-          {t("unsupported_browser_description", notSupported)}
-        </p>
+    <>
+      {showUnsupportedBrowserDialog && (
+        <AlertDialog
+          open={showUnsupportedBrowserDialog}
+          onOpenChange={setShowUnsupportedBrowserDialog}
+        >
```

Comment:
is this needed? 🤔 alert dialog's open prop should take care of it anyways right?

```suggestion
      {(
        <AlertDialog
          open={showUnsupportedBrowserDialog}
          onOpenChange={setShowUnsupportedBrowserDialog}
        >
```

Reply:
Yes, since it's rendering at the root, the performance gain is minimal. Relying on the open prop alone is enough. Removing it and pushing!